src/Core/Security/Application/UseCase/LoginOpenIdSession/LoginOpenIdSession.php

```diff
@@ -277,4 +276,160 @@ private function createResponse(?string $redirectUri, ?string $error = null): Lo
 
         return $response;
     }
+
+    /**
+     * Find User in Centreon or throw an exception
+     *
+     * @return ContactInterface
+     * @throws NotFoundException
+     */
+    private function findUserOrFail(): ContactInterface
+    {
+        $user = $this->provider->getUser();
+        if ($user === null) {
+            $this->info("User not found");
+            if (! $this->provider->canCreateUser()) {
+                throw new NotFoundException('User could not be created');
+            }
+            $this->info("Start auto import");
+            $this->provider->createUser();
+            $user = $this->provider->getUser();
```

Comment:
It is not the responsibility of the provider to read the user into the repository.